benchmarks/distributed/rpc/ps/launcher.py

```diff
@@ -0,0 +1,372 @@
+    if parameter_server_rank - configurations.trainer_count < rem:
+        ps_trainer_count += 1
+    return rpc.remote(
+        name,
+        ps,
+        args=(
+            parameter_server_rank,
+            ps_trainer_count,
+            *configured_args,
+        ),
+    )
+
+
+def run_trainer(
+    configurations, trainer_configurations, model, data, rank, ps_rref
+):
+    trainer_class = get_benchmark_trainer_map()[str(trainer_configurations["trainer_class"])]
+    configured_args = trainer_configurations["configurations"].values()
+    trainer = trainer_class(
+        rank,
+        configurations.trainer_count,
+        ps_rref,
+        *configured_args
+    )
+    trainer.train(model, data)
+    metrics = trainer.get_metrics()
+    return [rank, metrics]
+
+
+def call_trainers(configurations, trainer_configurations, model, train_data, parameter_server_rrefs):
+    futs = []
+    for trainer_rank in range(0, configurations.trainer_count):
+        trainer_name = get_name(
+            trainer_rank,
+            configurations.trainer_count,
+            configurations.parameter_server_count
+        )
+        ps_rref = None
+        if parameter_server_rrefs:
+            ps_rank = get_parameter_server_rank(trainer_rank, configurations)
+            ps_rref = parameter_server_rrefs[ps_rank]
+        fut = rpc.rpc_async(
+            trainer_name,
+            run_trainer,
+            args=(
+                configurations,
+                trainer_configurations,
+                copy.deepcopy(model),
+                train_data[trainer_rank],
+                trainer_rank,
+                ps_rref,
+            ),
+            timeout=configurations.rpc_async_timeout
+        )
+        futs.append(fut)
+    return futs
+
+
+def benchmark_warmup(
+    configurations, trainer_configurations, ps_configurations, model, data, parameter_server_rrefs
+):
+    if configurations.parameter_server_count > 0:
+        ps = get_benchmark_ps_map()[str(ps_configurations["ps_class"])]
+    futs = call_trainers(configurations, trainer_configurations, model, data, parameter_server_rrefs)
+    for fut in futs:
+        fut.wait()
+    for ps_rref in parameter_server_rrefs.values():
+        rpc.rpc_sync(
+            ps_rref.owner(),
+            ps.reset_state,
+            args=(ps_rref,)
+        )
+    print("benchmark warmup done\n")
+
+
+def split_list(arr, n):
+    return [arr[i::n] for i in range(n)]
+
+
+def get_ps_metrics(configurations, ps_configurations, parameter_server_rrefs):
+    rank_metrics = []
+    if configurations.parameter_server_count > 0:
+        ps = get_benchmark_ps_map()[str(ps_configurations["ps_class"])]
+        for rank, ps_rref in parameter_server_rrefs.items():
+            metrics = rpc.rpc_sync(
+                ps_rref.owner(),
+                ps.get_metrics_rpc,
+                args=(ps_rref,)
+            )
+            rank_metrics.append([rank, metrics])
```

Comment:
IIUC, this is not used in this PR? If yes, let's remove and add it when we need it.

Reply:
okay.